### ISSUE TITLE
Kernel 6.0.3 - SSL Error

### ISSUE THREAD

**Issue by EdwardAirs** [DisplayLink/evdi#385]
* Are you using the latest driver?
    
  * Are you using the latest EVDI version?
    
  * If you are using a DisplayLink device, have you checked 'troubleshooting'
    on DisplayLink's website?
    
  * Is this issue related to evdi/kernel?
    

Linux OS: POP_OS 22.04
Linux Kernel: 6.0.3
Using XOrg
DisplayLink - Latest

Dear Sir, 

Thank you for your effort.... 

With Kernel 5.19, 6.0.2, everything was working (Original DisplayLink modules, EVDI, etc), it updated to 6.0.3 and the EVDI broke, thus I am trying your solution (waiting for official patch from System76 .... or any OS developer).

Problem:

"sudo make install" 

Gives me the error SSL - file bss_file.c:67 not found:

make -C module install
make[1]: Entering directory '/home/usr_me/evdi/module'
make -C /lib/modules/6.0.3-76060003-generic/build M=$PWD INSTALL_MOD_PATH= INSTALL_MOD_DIR=/kernel/drivers/gpu/drm/evdi modules_install
make[2]: Entering directory '/usr/src/linux-headers-6.0.3-76060003-generic'
  INSTALL /lib/modules/6.0.3-76060003-generic//kernel/drivers/gpu/drm/evdi/evdi.ko
  SIGN    /lib/modules/6.0.3-76060003-generic//kernel/drivers/gpu/drm/evdi/evdi.ko
At main.c:167:
**- SSL error:FFFFFFFF80000002:system library::No such file or directory: ../crypto/bio/bss_file.c:67
- SSL error:10000080:BIO routines::no such file: ../crypto/bio/bss_file.c:75**
sign-file: ./certs/signing_key.pem
  DEPMOD  /lib/modules/6.0.3-76060003-generic
Warning: modules_install: missing 'System.map' file. Skipping depmod.
make[2]: Leaving directory '/usr/src_me/linux-headers-6.0.3-76060003-generic'
true
make[1]: Leaving directory '/home/usr_me/evdi/module'
make -C library install
make[1]: Entering directory '/home/usr_me/evdi/library'
install -d /usr/local/lib
install -m 755 libevdi.so.1.12.0 /usr/local/lib/libevdi.so.1.12.0
ln -sf libevdi.so.1.12.0 /usr/local/lib/libevdi.so.0
ln -sf libevdi.so.0 /usr/local/lib/libevdi.so
make[1]: Leaving directory '/home/usr_me/evdi/library'

Was following the issue #381 & #384 in particular @dominikogg 

Could you please address me what is my mistake?

**Comment by MiguelGuthridge** [comment]
I followed the steps in #384, and I got this error too. However, when I rebooted my laptop it worked perfectly.

**Comment by EdwardAirs** [comment]
I must of have done something wrong .... still wondering after 2 days ... I rebooted and my POP_OS just crashed and went to ZDM (the equivalent of GRUB for this system). 

Did a repair of the system that brought me back to Kernel 5.0.19, just followed the instructions on the DisplayLink web page, all working perfect.

Then I updated the system from the command line, it installed kernel 6.0.2 and 6.0.3, thus I was back to problem I posted here, rebooted , nothing, installed EVDI-DKMS 1.12.0 and all possible solutions like blacklisting UDL etc etc .... writing new config servers for the displays ,,,, rebooting again and again ... ZDM ..... 

Tried Ubutnu 22.04, Mint (can not remember the latest version), all with the same result .... wanted to try Gentoo but needed some sleep. 

Opted to go totally different route, RHEL, fresh install of Fedora 36, did all possible, nothing works, same as above ..... rebooted one last time ..... GRUB .... AGAIN ... .... but with a little surprise, I can choose between Kernel 6.0.5 of 5.0.17 ..... which I can Opt with Wayland or Xorg.... even Gnome X11 (this should be xorg?) or Gnome Classic (this should be Wayland?).... 

At the moment I am writing to you from my third screen using kernel 5.0.17 with gnome classic X11 with full Nvidia primary GPU , only 2 little snag it is a little slow to refresh where the mouse is and the laptop is a little hot, will try other options, and combinations .... installing specific kernel 5.0.19 (like POP_OS that was working as if it was native in the system like windows) ... then will be monitoring and waiting, testing other kernels till i get the latest one working as 5.0.19, till then I am sticking with wat works because can not afford to play around, as I use my system for work .... here we can open a huge debate.... 

In conclusion, after kernel 5.0.19 there is some mighty problem, with the DisplayLink - EVDI - DKMS modules, it is not Nvidia, it is not INTEL, it is not the hardware .... the brand of the docking station or the laptop/desktop .... there is something new/broken in the new kernels ... will soon be trying CentoOS as the most stable in field since fedora will be updated very soon.... 

This is the only solution I found and the problems I encountered .... As soon as somebody has a valid alternative, official would be best as there is a bunch of us that are not programmers (I personally feel a nob still learning every day) but love Linux and use it for work. 

Sorry for the ramble .... just tired.

**Comment by MiguelGuthridge** [comment]
It's working for me on Fedora 36.

```
$ uname -srm       
Linux 6.0.5-200.fc36.x86_64 x86_64
$ . /etc/os-release
$ echo $NAME
Fedora Linux
$ echo $VERSION_ID
36
```

**Comment by EdwardAirs** [comment]
...... what just happened? 

I was pasting the 5.19.6 kernel output to show you .... rebooted thus to show you the 6.05 out put but .... 

I got 3 monitors working .... possibly it just needed to settle and embed the installation .... read somewhere sometimes the Nvidia drivers need some reboots to settle in correctly ... 

Anyhow: 
$ uname -srm
Linux 6.0.5-200.fc36.x86_64 x86_64
$ . /etc/os-release
$ echo $NAME
Fedora Linux

Mr. Miguel Guthridge, thank you my kind sir. 
Thank you to all the develops behind this patch ... thank you to all. ....

**Comment by jtmackoy** [comment]
Hey @EdwardAirs - wanted to share something I learned that may have impacted your original issue on an Ubuntu kernel.  See [my closing comment](https://github.com/DisplayLink/evdi/issues/387#issuecomment-1306544312) on #387.

I know you're cruising along on a Fedora kernel at the moment.  Maybe it'll help in the future if you ever run across this again.

If my experience using DisplayLink drivers on Ubuntu is any indication, it may very well be.  :smiley:

**Comment by EdwardAirs** [comment]
> Hey @EdwardAirs - wanted to share something I learned that may have impacted your original issue on an Ubuntu kernel. See [my closing comment](https://github.com/DisplayLink/evdi/issues/387#issuecomment-1306544312) on #387.
> 
> I know you're cruising along on a Fedora kernel at the moment. Maybe it'll help in the future if you ever run across this again.
> 
> If my experience using DisplayLink drivers on Ubuntu is any indication, it may very well be. smiley

"I did Learn something new today " .... Indeed I did, every day. 

Thank you for the update. 

Yes cruising with kernel 6.0.5 ... a happy bunny .... only thanks to the whole comunity!!!
